script/app-coverage-report.js

```diff
@@ -27,34 +41,67 @@ const printCoverage = coverageResults => {
   console.log(coverageTable.toString());
 };
 
-const generateCoverage = (rootDir, coverageSummary) =>
-  Object.keys(coverageSummary).reduce((acc, coverageResult) => {
+const generateCoverage = (rootDir, coverageSummary) => {
+  // Get application manifests & entry names
+  const manifests = getAppManifests(path.join(__dirname, '../'));
+  const entryNames = manifests.map(manifest => manifest.entryName);
+
+  // Set initial coverage object
+  const initialCoverage = Object.assign(
+    ...entryNames.map(entryName => ({
+      [entryName]: {
+        path: manifests
+          .find(obj => obj.entryName === entryName)
+          .entryFile.replace(rootDir, ''),
+        lines: { total: 0, covered: 0, skipped: 0, pct: 0 },
+        statements: { total: 0, covered: 0, skipped: 0, pct: 0 },
+        functions: { total: 0, covered: 0, skipped: 0, pct: 0 },
+        branches: { total: 0, covered: 0, skipped: 0, pct: 0 },
+      },
+    })),
+  );
+
+  return Object.keys(coverageSummary).reduce((acc, coverageResult) => {
     const appName = coverageResult.replace(rootDir, '').split(path.sep)[0];
-    // If appName exists, add coverageResult to overall coverage
+
     if (appName) {
-      if (acc[appName]) {
+      // Find coverage if it exists by checking if the path of the coverage result file
+      // includes an application path, e.g. 'vre/chapter36/'
+      const resultFilePath = coverageResult.replace(rootDir, '');
+      const coverageItem = Object.values(acc).find(appCov =>
+        resultFilePath.includes(
+          `${appCov.path.substring(0, appCov.path.lastIndexOf('/'))}/`,
+        ),
+      );
+
+      // Get key of coverageItem
+      const coverageKey = Object.keys(acc).find(
+        key => acc[key] === coverageItem,
+      );
+
+      // Update coverage
+      if (acc[coverageKey]) {
```

Comment:
I also think that we could just use `coverageItem` instead of doing the extra steps required to get `coverageKey`.  An example:

```js
    // Average coverageResult with respective acc segment if present
    // Each 'seg' represents coverage by line, function, statement, and branch
    Object.keys(coverageItem || {})
      .forEach(seg => {
        coverageItem[seg].total += coverageSummary[coverageResult][seg].total;
        // ...
      });
```

Reply:
Yeah that definitely helps clean things up! 👍